trick_sims/SIM_fluid/models/fluid/graphics/src/marching_cubes.cpp

```diff
@@ -0,0 +1,102 @@
+        geometryCaseIndex |= 64;
+    if (grid.isoValues[7] > isoLevel)
+        geometryCaseIndex |= 128;
+
+    // no edges in grid cell contain a triangle vertex
+    if(edgeTable[geometryCaseIndex] == 0)
+        return;
+    
+    glm::vec4 origin = glm::vec4(0, 0, 0, 1);
+    // On edges that contain a triangle vertex, interpolate that triangle vertex based on the isovalues of the endpoint vertices of that edge
+    if (edgeTable[geometryCaseIndex] & 1)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[0], grid.vertices[1], grid.isoValues[0], grid.isoValues[1]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 2)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[1], grid.vertices[2], grid.isoValues[1], grid.isoValues[2]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 4)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[2], grid.vertices[3], grid.isoValues[2], grid.isoValues[3]));
+    else 
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 8) 
+        vertices.push_back(vlerp(isoLevel, grid.vertices[3], grid.vertices[0], grid.isoValues[3], grid.isoValues[0]));
+    else 
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 16)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[4], grid.vertices[5], grid.isoValues[4], grid.isoValues[5]));
+    else 
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 32)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[5], grid.vertices[6], grid.isoValues[5], grid.isoValues[6]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 64)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[6], grid.vertices[7], grid.isoValues[6], grid.isoValues[7]));
+    else 
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 128)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[7], grid.vertices[4], grid.isoValues[7], grid.isoValues[4]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 256)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[0], grid.vertices[4], grid.isoValues[0], grid.isoValues[4]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 512)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[1], grid.vertices[5], grid.isoValues[1], grid.isoValues[5]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 1024)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[2], grid.vertices[6], grid.isoValues[2], grid.isoValues[6]));
+    else
+        vertices.push_back(origin);
+    if (edgeTable[geometryCaseIndex] & 2048)
+        vertices.push_back(vlerp(isoLevel, grid.vertices[3], grid.vertices[7], grid.isoValues[3], grid.isoValues[7]));
+    else 
+        vertices.push_back(origin);

```

Comment:
Just an idea. Benefits are that if there is a mistake, you only have to change it in one place instead of 12, for example. Also less code to scroll through.

Reply:
Resolved in 
https://github.com/nasa/trick/pull/1170/commits/730c320172ef25aec9eb65265ad2b2bb9440a888